backend/fms_core/template_importer/row_handlers/library_qc/library_qc.py

```diff
@@ -74,8 +73,15 @@ def process_row_inner(self, sample_container, measures, process, process_measure
 
         # library size
         library_size = measures['library_size']
-        if library_size is None:
-            self.errors['library_size'] = 'Library size must be specified'
+        if not source_sample_obj.is_pool:
```

Comment:
if you have a pooled sample (top up) that you converted to a library it won't have a library_size and we will not be able to set one with this test... Maybe add an "or all(derived_sample.library and derived_sample.library.library_size is None for derived_sample in source_sample_obj.derived_samples)"